cmd/clusterctl/cmd/config_provider.go

```diff
@@ -187,8 +187,10 @@ func printComponents(c client.Components, output string) error {
 		if _, err := os.Stdout.Write(yaml); err != nil {
 			return errors.Wrap(err, "failed to write yaml to Stdout")
 		}
-		os.Stdout.WriteString("\n")
-		return err
+		if _, err := os.Stdout.WriteString("\n"); err != nil {
+			return errors.Wrap(err, "failed to write yaml to Stdout")
```

Comment:
Nit, while this is unlikely to ever actually happen, the error on this line is identical to 188, that could make it slightly harder to debug if you just end up with an error and not a line number the error was generated from. Might be better to change this slightly so it is a unique error, WDYT?

Reply:
Yup, will do